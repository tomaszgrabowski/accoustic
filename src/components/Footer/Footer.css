footer{
    display: flex;
    padding: 2rem;
    justify-content: space-between;
}
