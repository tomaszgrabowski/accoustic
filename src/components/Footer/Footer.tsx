import React from 'react';
import Date from '../Date/Date';
import './Footer.css';

interface IProps {
    author: string;
    date: string;
}

const Footer = ( props: IProps ) => {
    return (
        <footer>
            <span>{ props.author }</span>
            <Date date={ props.date } format={ 'DD-mm-YYYY HH:mm' }/>
        </footer>
    );
};

export default Footer;
