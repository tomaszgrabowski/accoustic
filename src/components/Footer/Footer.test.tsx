import { shallow } from 'enzyme';
import * as faker from 'faker';
import React from 'react';
import renderer from 'react-test-renderer';
import Footer from './Footer';

describe( 'Footer', () => {
    test( 'given author and date when render then show author and date', () => {
        const author = `${ faker.name.firstName() } ${ faker.name.lastName() }`;
        const date = faker.date.recent().toISOString();
        const wrapper = shallow( <Footer author={ author } date={ date }/> );
        expect( wrapper.exists( 'footer' ) ).toBeTruthy();
        expect( wrapper.find( 'span' ).text() ).toEqual( author );
        expect( wrapper.exists( 'Date' ) ).toBeTruthy();
    } );
    
    test( 'given Footer component when render then renders properly', () => {
        const wrapper = renderer.create( <Footer author={ 'author' } date={ '2012-01-01T23:23:23Z' }/> ).toJSON();
        expect( wrapper ).toMatchSnapshot();
    } );
} );
