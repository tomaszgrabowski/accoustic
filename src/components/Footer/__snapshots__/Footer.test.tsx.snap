// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer given Footer component when render then renders properly 1`] = `
<footer>
  <span>
    author
  </span>
  <small>
    02-23-2012 00:23
  </small>
</footer>
`;
